nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -441,13 +441,6 @@ def _greedy_decode(
                 # If blank token is predicted, exit inner loop, move onto next timestep t
                 if k == self._blank_index:
                     not_blank = False
-
```

Comment:
Why are these being moved ?

Reply:
Ok so the `while` loop operates until the `blank` token is predicted or predictions in the considered frame exceed `max_symbols`.
In the first case, the algorithm sets `not_blank = False` (thus, this is the final `while` iteration) and appends new lists to the `alignments` and `frame_confidence`.
But in the second case (`blank` is not predicted, `max_symbols` will be exceeded), the loop just finishes without adding the new lists. This leads to adding the next frame prediction alignment and confidence to the last existing lists, causing shifted alignment and the issue in question.
So, moving the appendment of new lists to the end of the loop is equivalent to the source implementation for the first case and fixes the second.

Reply:
I don't have the time to test this, but someone has to test that alignments from first case indeed match this new version before merging this